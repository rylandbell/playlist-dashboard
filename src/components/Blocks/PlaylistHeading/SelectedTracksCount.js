import React from 'react';
import { getPlaylistDuration } from '../../../helper.js';

const SelectedTracksCount = ({ tracks, filteredTracks }) => {
  //only count tracks with non-null features data
  const allTracksCount = tracks.filter(track => !!track).length;
  const filteredTracksCount = filteredTracks.length;
  const duration = getPlaylistDuration(filteredTracks);

  let durationPhrase = '';
  if (duration.hr && duration.hr > 0) {
    durationPhrase += `${duration.hr} hr `;
  }
  durationPhrase += `${duration.min} min`;

  return (
    <div>
      <span className="playlist-heading__tracks-count">{`${filteredTracksCount} of ${allTracksCount} songs, ${durationPhrase}`}</span>
    </div>
  );
};

export default SelectedTracksCount;
