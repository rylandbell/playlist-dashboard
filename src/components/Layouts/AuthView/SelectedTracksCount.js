import React, { Component } from 'react';

class ExporeTracks extends Component {
  render() {
    const getPlaylistDuration = tracksArray => {
      const total_ms = tracksArray.reduce(
        (accumulator, track) => {
          return accumulator + track.track.duration_ms;
        },
        0
      );

      const total_min = Math.round(total_ms / (1000 * 60));

      const duration = {
        hr: Math.floor(total_min / 60),
        min: Math.floor(total_min % 60)
      }
      return duration;
    }

    //only count tracks with non-null features data
    const allTracksCount = this.props.featuresData.filter(track => !!track).length;
    const filteredTracksCount = this.props.filteredTracks.length;
    const duration = getPlaylistDuration(this.props.filteredTracks);

    let durationPhrase = "";
    if (duration.hr && duration.hr > 0) {
      durationPhrase += `${duration.hr} hr `;
    }
    durationPhrase += `${duration.min} min`;

    return (
      <div>
        <span className="auth-view__selected-tracks-count">{`${filteredTracksCount} of ${allTracksCount} songs, ${durationPhrase}`}</span>
      </div>
    );
  }
}

export default ExporeTracks;

