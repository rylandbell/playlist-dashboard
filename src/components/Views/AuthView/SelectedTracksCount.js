import React, { Component } from 'react';

class ExporeTracks extends Component {
  render() {
    //only count tracks with non-null features data
    const allTracksCount = this.props.featuresData.filter(track => !!track).length;
    const filteredTracksCount = this.props.filteredTracks.length;

    return (
      <div className="row auth-view__selected-tracks-count-wrapper">
        <div className="auth-view__selected-tracks-count pull-right">{`${filteredTracksCount} of ${allTracksCount} Selected`}</div>
      </div>
    );
  }
}

export default ExporeTracks;

