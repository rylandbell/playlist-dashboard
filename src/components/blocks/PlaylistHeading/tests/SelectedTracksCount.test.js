import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import SelectedTracksCount from "../SelectedTracksCount";
import { testFilteredTracks, testTracks } from "./test-data";

describe("SelectedTracksCount component", () => {
  const testProps = {
    filteredTracks: testFilteredTracks,
    tracks: testTracks
  };

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<SelectedTracksCount {...testProps} />, div);
  });

  it('shallow render matches snapshot', () => {
    const tree = toJson(shallow(<SelectedTracksCount {...testProps} />));
    expect(tree).toMatchSnapshot();
  });
});
