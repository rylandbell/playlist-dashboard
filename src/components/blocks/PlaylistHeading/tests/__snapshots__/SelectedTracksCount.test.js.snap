// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedTracksCount component shallow render matches snapshot 1`] = `
<div>
  <span
    className="playlist-heading__tracks-count"
  >
    2 of 3 songs, 6 hr 46 min
  </span>
</div>
`;
